### PR TITLE
feat(events): use community image as fallback when event image is missing #288

### PR DESCRIPTION
**Pull request by KOWSIK-M**
### Event Image
- [X] If an event has its own image, that image is displayed. 
- [X] If an event has no image but the related community has one, the community image is displayed.
- [X] If neither the event nor the community has an image, the placeholder image is displayed.
### Consistency
- [X] The logic applies consistently wherever event images are shown (event cards, detail views, etc.).
### Testing
- [x] Manual testing is performed successfully on both desktop and mobile devices.
- [X] Full automated test coverage is implemented for the added code.
- [X] All automated tests pass.

# Major Code changes :

- In Pom.xml, added a new dependency **mockwebserver** to mock HTTP endpoints in tests.
 > This is optional and you can remove it, but I updated DownloadUtilTest.java code with this dependency
 the dependency I added is at line **282** 
 ```
 <dependency>
    <groupId>com.squareup.okhttp3</groupId>
    <artifactId>mockwebserver</artifactId>
    <version>4.12.0</version>
    <scope>test</scope>
</dependency>
```
---------
## Other changes: 
| File | Changes|
|:--- | ---:|
| **EventService.java** | Logic to handle image fallback |
| **EventServiceIT.java**| New integration test added |
| **ImageUtil.java** |  Improved handling of ImageDto & content type |
| **DownloadUtilTest.java** | Expanded tests to cover edge cases & error handling |
| **ImageUtilIT.java** | Added integration tests for ImageUtil |

- Small fixes in **ConfirmationView**, **DemoDataCreatorJsonIT**, and **ParticipationServiceIT**.

#### Extras : 
- [ ] update DemoDataCreator for better manual testing.
   Reason : I couldn't handle errors in Tests.

### CODE REVIEW

File: src/main/java/app/komunumo/util/ImageUtil.java [hunk 1]
```diff
@@ -58,14 +58,16 @@ public static void initialize(final @NotNull AppConfig appConfig) {
         uploadImagePath = appConfig.files().basedir().resolve(RELATIVE_IMAGE_PATH);
     }
 
-    public static @Nullable String resolveImageUrl(final @Nullable ImageDto image) {
+    public static @NotNull String resolveImageUrl(final @Nullable ImageDto image) {
         if (image == null || image.id() == null) {
-            return null;
+            return "/images/komunumo.svg"; // placeholder
```
[the remark this answers is not in the document]
Comment (KOWSIK-M) [reply src/main/java/app/komunumo/util/ImageUtil.java:63]: Thank you for the clarification, sir. I misunderstood the purpose of the default /images/komunumo.svg — I assumed it was a placeholder to show when no image was present. I now understand it’s actually the default instance logo, and placeholders are dynamically generated later. I’ll revert that change and ensure the method only returns null when the image is missing, so the correct dynamic placeholder logic remains intact.

File: src/test/java/app/komunumo/data/demo/DemoDataCreatorJsonIT.java [hunk 2]
```diff
@@ -43,6 +43,8 @@ void testDemoDataImporterWithJson() {
         page.waitForSelector("h1:has-text('Komunumo Test')");
         captureScreenshot("home-page-with-demo-data");
 
+        page.waitForSelector("vaadin-card");
```
Comment (McPringle) [inline src/test/java/app/komunumo/data/demo/DemoDataCreatorJsonIT.java:46]: This change is okay, but it would be better in a separate PR because it does not belong to #288.

File: src/test/java/app/komunumo/data/service/ParticipationServiceIT.java [hunk 3]
```diff
@@ -107,7 +107,7 @@ void joinEventAnonymousUser() {
         assertThat(participations).hasSize(1);
 
         final var user = userService.getUserByEmail(email).orElseThrow();
-        assertThat(user).isNotNull();
+        assertThat(user.email()).isEqualTo(email);
```
Comment (McPringle) [inline src/test/java/app/komunumo/data/service/ParticipationServiceIT.java:110]: Again, this change is okay, but it would be better in a separate PR because it does not belong to #288.

File: src/test/java/app/komunumo/util/ImageUtilIT.java [hunk 4]
```diff
@@ -41,55 +41,60 @@ class ImageUtilIT extends IntegrationTest {
...
+        final String prefix1 = id.substring(0, 2);
+        final String prefix2 = id.substring(2, 4);
+        return prefix1 + java.io.File.separator + prefix2;
+    }
+
     @Test
     void resolveImageUrl() {
-        assertThat(ImageUtil.resolveImageUrl(null)).isNull();
+        // null should give placeholder now
+        assertThat(ImageUtil.resolveImageUrl(null)).isEqualTo("/images/komunumo.svg");
```
Comment (McPringle) [inline src/test/java/app/komunumo/util/ImageUtilIT.java:54]: This is not the placeholder. We can't set the placeholder in this method because the placeholder URL contains the width and size. The placeholder is dynamically created using the `PlaceholderImageGenerator`.
